### PR TITLE
My enumerables

### PR DESCRIPTION
**Pull request by teekaytech**
# Hello,

### In this project, I successfully completed the cloning of some methods in Ruby Enumerables as required. Here are the highlights of the activities completed:

- I implemented my own versions of each, each_with_index, select, all, any, none, count, map, and inject methods.
- I added my methods to the existing enumerables module
- I test my_inject method with the multiply_els([2,3,4,5]) method.
- I utilize yield() and call() methods.
- I used rubocop linter to validate and cross-check for possible errors.
- I used Git Flow in this project, for the first time.

### Thank you.

### CODE REVIEW

File: main.rb
Comment (suretrust) [inline main.rb]: * [x] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5,6,7].my_all?(Integer) != [1,2,3,4,5,6,7].all?(Integer)

['saheed', 'oladele', 'suretrust'].my_all?(/o/) != ['saheed', 'oladele', 'suretrust'].all?(/o/)

[1,2,3,4,5,6,7].my_all?(3) != [1,2,3,4,5,6,7].all?(3)
```

File: main.rb
Comment (suretrust) [inline main.rb]: * [x] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
['saheed', 'oladele', 'suretrust'].my_any?(/d/) != ['saheed', 'oladele', 'suretrust'].any?(/d/)

['cat', 'dog'].my_any?('cat') != ['cat', 'dog'].any?('cat')
```

File: main.rb
Comment (suretrust) [inline main.rb]: * [x] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
['saheed', 'oladele', 'suretrust'].my_none?(/d/) != ['saheed', 'oladele', 'suretrust'].none?(/d/)

['saheed', 'oladele', 'suretrust'].my_none?(5) != ['saheed', 'oladele', 'suretrust'].none?(5)
```

File: main.rb
Comment (suretrust) [inline main.rb]: * [x] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
array = (0..100).to_a

array.my_inject(:+) != array.inject(:+)
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @teekaytech, 👋 

### Great job thus far! 🙌 However, some of your methods need more work. 👇 

- [ ] Please see the test cases below and fix them as appropriate.

**Kindly submit for another code review request once this has been fixed.** _Happy coding!_ 🎉

File: main.rb
Comment (teekaytech) [reply main.rb]: Fixed.

File: main.rb
Comment (teekaytech) [reply main.rb]: Fixed.

File: main.rb
Comment (teekaytech) [reply main.rb]: Fixed.

File: main.rb
Comment (teekaytech) [reply main.rb]: Fixed.

**Comment by teekaytech** [comment]
## In this version of my pull request, I have successfully fixed all changes as listed out by the previous TSE.
Kindly review as I wait for your approval.
Thank you.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Great work @teekaytech :+1:. You're almost there**
- When a pattern other than a regex or class is given to my_none?, it should return true **only** if none of the elements matches the pattern
Example `["dog" "door" "rod" "blade", 5].my_none?(5)` should return `false` instead of `true`

- my_inject is still not in order.Consider this:
`p [10, 5, 20, 14, 8, 8].my_inject { | acc, x | acc + x }` cannot be 10 but your method returns 10.

- The multiply_els method is missing in your file.




_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by teekaytech** [comment]
# Thank you, Moses. 

I have reviewed the issues raised by the previous TSE and updated my codebase. 

## Kindly review my project for approval.
### Thank you.

**Review by mosesogwo — APPROVED** [review]
**Hi @teekaytech:clap: You have done well:+1: Your project is approved.**

However, take note of this: inject should have the ability to be used to find the longest word as shown in the [ruby documentation](https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-inject): 
```
# find the longest word

longest = %w{ cat sheep bear }.inject { |memo, word| memo.length > word.length ? memo : word }
```

Your my_inject currently throws an error, but since we haven't pointed this out before now, I'll let you look at this on your own.

**Good luck in your next project :stars:**

### Status: Approved :heavy_check_mark:

**Pull request merged**
